### ISSUE TITLE
Updated License Agreement

### ISSUE THREAD

**Issue by cturner35** [fastlane/fastlane#13619]
### Question Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Question Subject
 Apple Developer Program License Agreement

#### Question Description
Is there a way to tell if an account that I'm a team member on, needs to have the agent logged in to accept a new Apple license agreement? Is anything like this even exposed by the Apple api or can this only be done by screen scraping? 

Basically, I need to be able to tell if this message is displayed when I login to the developer portal or app store connect. "The Apple Developer Program License Agreement has been updated."

**Comment by janpio** [comment]
I remember similar issues in the past. Have you searched the fastlane issues for any previous responses?

**Comment by cturner35** [comment]
Yes, I did try to find anything I could here on the issues and also googling but couldn't find anything.

One thing we learned is this, once a the app store connect account becomes locked then if you try to access the apps array in spaceship it will throw an error. We've basically just used a begin/rescure to determine if the account is locked and needs to accept an updated license. There is an array of apps that when accessed throws an exception, heres a sample of our code:

portal = Spaceship::Portal.login(username, password)
portal.team_id = {team_id}
portal.apps

When the portal.apps line is executed it throws the exception and seems to work for what we need. 

I was wondering if there was an easier way to do this? Or if there is a way to tell that the team agent needs to accept the new terms before the account is locked?

**Comment by janpio** [comment]
Could you maybe refresh my memory how the "update licence agreement", lock account process works exactly?

---

> Basically, I need to be able to tell if this message is displayed when I login to the developer portal or app store connect. "The Apple Developer Program License Agreement has been updated."

Hm, there is code and tests that indicate this should be a feature already:

https://github.com/fastlane/fastlane/blob/16b362809888d5bf57d68162b0639d2ccbebd09e/spaceship/lib/spaceship/client.rb#L48
https://github.com/fastlane/fastlane/blob/16b362809888d5bf57d68162b0639d2ccbebd09e/spaceship/lib/spaceship/client.rb#L638-L640
https://github.com/fastlane/fastlane/blob/9d370e77f35cc509442a3ab62d51394bfecbf729/spaceship/spec/client_spec.rb#L77-L84

**Comment by cturner35** [comment]
What I've found from trial and error is this ...

First, we get an email when a team has to go an accept the new terms. When you login you'll see a yellow banner and that will stay that way for somewhere around ten days to two weeks? Not sure b/c I just learned this last time the updated. After that window the account becomes locked and we can't create certs, profiles, etc until the terms are accepted. At this point that yellow banner is now red too and calling the .apps property throws an exception.

I had seen those references before but I didnt understand how to call that method. Heh, I'm new to Ruby :/

**Comment by janpio** [comment]
`spaceship` usually takes care of that internally, when parsing responses: https://github.com/fastlane/fastlane/blob/16b362809888d5bf57d68162b0639d2ccbebd09e/spaceship/lib/spaceship/client.rb#L594-L608

What actions or commands did you have problems with using when the account was in the "blocked" state? Maybe those didn't actually use spaceship?

**Comment by cturner35** [comment]
Ahhh ok that makes sense then about why I couldnt find it. 

We need a way to go through all of the teams we are have been added to and be able to tell if each team needs to accept the new terms.

The problem was what we ended up using with the .apps doesnt throw the exception until the timeframe is up and the red banner is displayed. 

Hopefully I'm making sense :)

**Comment by janpio** [comment]
So you basically need a function to check if a new agreement has to be accepted _without_  actually doing any changes on an app?

**Comment by cturner35** [comment]
Correct. We publish white label apps for our customers and we need a way to tell if that customer needs to accept the new agreement so we can update their app.

**Comment by janpio** [comment]
Okay, so one way would be to write an lane/action/plugin that uses one of the methods that could trigger this code, but doesn't actually change anything. The code you posted above seems to just request a list of the apps, is this implemented in fastlane as an action maybe?

**Comment by cturner35** [comment]
The .apps call is implemented in a file that is executed by a fastlane lane.

So, the code I pasted above, will cause the error to be thrown (for whatever reason) but my problem is that the exception is thrown only once that time window has closed. I was wondering if there was a way to tell before the window closes, ie, when you login, the banner is yellow.

**Comment by janpio** [comment]
Ah, got it.

From my naive reading of the code, it should if it is returned as a string for a call to the server - but I have no idea what kind of requests this makes and if any of these actually return that information.

To find out, one will need such an account it seems. Then create a fastfile that does some non-updating actions on the account and see what comes back, maybe by inspecting the http traffic...

**Comment by cturner35** [comment]
Ahh good point and thanks for the help!

When I first posted this, I thought our work around was broken b/c no exceptions where being thrown when the banner was yellow. Then I ran our script again a couple weeks later and got all the errors. For now its good enough to know after that window is closing but if I get time I'll def try your suggestion of looking into that response from the server.

Thanks!
